tests/_utils/kafka.py

```diff
@@ -0,0 +1,152 @@
+# SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import subprocess
+import time
+import typing
+from collections import namedtuple
+from functools import partial
+
+import pytest
+
+if (typing.TYPE_CHECKING):
+    from kafka import KafkaConsumer
+
+# Amount of seconds we will wait for the Kafka consumer to be assigned partitions
+PARTITION_ASSIGNMENT_TIMEOUT = 15
+
+# actual topic names not important, but we will need two of them.
+KafkaTopics = namedtuple('KafkaTopics', ['input_topic', 'output_topic'])
+
+# pylint: disable=invalid-name
+_kafka_consumer = None
+kafka_server = None
+zookeeper_proc = None
+# pylint: enable=invalid-name
+
+
+@pytest.fixture(name='kafka_topics', scope='function')
+def kafka_topics_fixture():
+    yield KafkaTopics(f'morpheus_unittest_input_{time.time()}', f'morpheus_unittest_output_{time.time()}')
```

Comment:
Is `time` the best discriminator we can use? This has the potential to create a lot of topics if we run tests repeatedly without restarting kafka. What are your thoughts on using something like this?
```python
import os
import re
def get_test_name_for_kafka_topic():
    test_name = os.environ.get("PYTEST_CURRENT_TEST")
    test_name = re.sub("[:/\[\]\(\)]", "-", test_name)
    test_name = re.sub(",", "_", test_name)
    return test_name
```
which produces something along these lines:
```python
"tests-test_kafka_oauth.py--test_kafka_oauth-use_cpp- -call-"
"tests-test_kafka_oauth.py--test_kafka_oauth-use_python- -call-"
```